module/extension/Director/src/Director.cpp

```diff
@@ -110,16 +118,23 @@ namespace module::extension {
         if (it != providers.end()) {
             auto provider = it->second;
 
-            // Can't add causing to a Start or Stop
-            if (provider->classification == ProviderClassification::STOP
-                || provider->classification == ProviderClassification::START) {
+            // Can't add when to a Start or Stop
+            if (provider->classification == Provider::Classification::STOP
+                || provider->classification == Provider::Classification::START) {
                 throw std::runtime_error("You cannot use the 'When' DSL word with Start or Stop.");
             }
 
+            // If the initial read throws an exception then it starts as false
+            bool current = false;
+            try {
+                // This throws an exception when the state hasn't been emitted yet
+                current = when.validator(when.current());
+            }
+            catch (...) {
```

Comment:
Empty catch everything feels a little dirty. Can we log the exception? is that worth doing?

Reply:
The purpose of this is that if you are trying to check a when condition and the thing you're checking doesn't exist you assume you can't run.

Reply:
There is one on line 128